torch/_dynamo/variables/tensor.py

```diff
@@ -498,7 +498,43 @@ def make_const_size_variable(x, **options):
                 example_value=None,
                 **options,
             )
-        elif name in ("tolist", "backward", "data_ptr"):
+        elif name == "tolist":
+            if free_symbols(self.size):
+                unimplemented(f"Tensor.{name}")
```

Comment:
Yeah, we could.

Reply:
IMO, it's better, as it will ensure tolist() works with export with no constraints (whereas this will always fail, even though a static annotation would make the export pass)